src/osp/Universe.h

```diff
@@ -29,24 +29,23 @@
 
 #include <entt/entity/registry.hpp>
 
-#include <vector>
 #include <cstdint>
+#include <tuple>
+#include <vector>
+
 
 namespace osp::universe
 {
 
 class ISystemTrajectory;
 class ITypeSatellite;
 
-//using Satellite = entt::entity;
-//ENTT_OPAQUE_TYPE(Satellite, uint32_t);
-enum class Satellite: entt::id_type {};
-
-
-using MapSatType = std::map<std::string, std::unique_ptr<ITypeSatellite>>;
+enum class Satellite : entt::id_type {};
 
-//typedef uint64_t Coordinate[3];
-//typedef Math::Vector3<int64_t> Coordinate;
+enum class TypeSatIndex : uint16_t
+{
+    Invalid = std::numeric_limits<uint16_t>::max()
```

Comment:
no more than 16 thousand satellites?

Reply:
Only 16 thousand different types of satellites. All I can think of is Planet, Vehicle, ActiveArea, Star, BlackHole, Asteroid, and I doubt anyone will be able to think about a thousand different more.

Reply:
Then reduce it to uint8_t from uint16_t.

I doubt there's a meaningful difference between more than 100 different types of celestial bodies, much less 255.